AdyenActions/Components/SDK/TwintSDKActionComponent.swift

```diff
@@ -0,0 +1,206 @@
+            context: AdyenContext,
+            configuration: Configuration,
+            twint: Twint
+        ) {
+            self.context = context
+            self.configuration = configuration
+            self.twint = twint
+        }
+
+        /// Handles TwintSDK action.
+        ///
+        /// - Parameter action: The Twint SDK action object.
+        public func handle(_ action: TwintSDKAction) {
+            AdyenAssertion.assert(message: "presentationDelegate is nil", condition: presentationDelegate == nil)
+            twint.fetchInstalledAppConfigurations { [weak self] installedApps in
+                guard let self else { return }
+                
+                guard let firstApp = installedApps.first else {
+                    let errorMessage = localizedString(
+                        LocalizationKey.twintNoAppsInstalledMessage,
+                        self.configuration.localizationParameters
+                    )
+                    self.handleShowError(errorMessage)
+                    return
+                }
+                
+                if installedApps.count > 1 {
+                    self.presentAppChooser(for: installedApps, action: action)
+                } else {
+                    self.invokeTwint(app: firstApp, action: action)
+                }
+            }
+        }
+
+        private func invokeTwint(app: TWAppConfiguration, action: TwintSDKAction) {
+
+            let error = twint.pay(
+                withCode: action.sdkData.token,
+                appConfiguration: app,
+                callback: configuration.callbackAppScheme
+            )
+            
+            if let error {
+                handleShowError(error.localizedDescription)
+                return
+            }
+            
+            RedirectListener.registerForURL { [weak self] url in
+                self?.twint.handleOpen(url) { [weak self] error in
+                    self?.handlePaymentResult(error: error, action: action)
+                }
+            }
+        }
+
+        private func presentAppChooser(for installedApps: [TWAppConfiguration], action: TwintSDKAction) {
+            let appChooserViewController = twint.controller(
+                for: installedApps,
+                selectionHandler: { [weak self] in
+                    self?.invokeTwint(
+                        app: $0 ?? installedApps[0],
+                        action: action
+                    )
+                },
+                cancelHandler: { [weak self] in
+                    guard let self else { return }
+                    self.delegate?.didFail(
+                        with: ComponentError.cancelled,
+                        from: self
+                    )
+                }
+            )
+            
+            if let viewController = appChooserViewController, let delegate = presentationDelegate {
+                present(viewController, presentationDelegate: delegate)
+            }
```

Comment:
Maybe add "else" with logged warning? Otherwise nothing will happen and merchant will have no clue why..